### PR TITLE
[MIG][13.0] base_gs1_barcode: Migration to v13.0

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT22176

### CODE REVIEW

**Review by jimhoefnagels — APPROVED** [review]
LG

File: base_gs1_barcode/models/gs1_barcode.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    data_type = fields.Selection(
+        [
+            ("string", "Any character string"),
+            ("numeric", "Numeric value"),
+            ("date", "Date"),
+        ],
+        "Data Type",
+        default="string",
+        required=True,
+        oldname="type",
```
Comment (pedrobaeza) [inline base_gs1_barcode/models/gs1_barcode.py]: Please remove this for avoiding runbot warning

Comment (sergio-teruel) [reply base_gs1_barcode/models/gs1_barcode.py]: Ok. Done!!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**